demos/scalar_advection/scalar_advection.py

```diff
@@ -1,6 +1,6 @@
 # Demo for pure scalar advection - this is adapted from the Firedrake DG_advection demo,
```

Comment:
@w3168 pointed out that this case should not be in demos. What do we do with it?